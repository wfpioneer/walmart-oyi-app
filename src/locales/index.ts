import * as RNLocalize from 'react-native-localize';
import I18n from 'i18n-js';
// @ts-ignore
import moment from 'moment';
import 'moment/locale/es';
import en from './en';
import es from './es';
import zh from './zh';

export const setI18nConfig: () => void = () => {
  // fallback if no available language fits
  const fallback = { languageTag: 'en', isRTL: false };

  const { languageTag } = RNLocalize.findBestAvailableLanguage(['en', 'es', 'zh'])
    || fallback;

  // set i18n-js config
  I18n.translations = { en, es, zh };
  I18n.locale = languageTag;

  // the following is to set up moment to work with spanish localization
  moment.locale(languageTag);
};

export const setLanguage = (language: string) => {
  if (language !== 'en' && language !== 'es' && language !== 'zh') {
    throw new Error('Unsupported language');
  }

  I18n.locale = language;
  moment.locale(language);
};

export const strings: (
  tag: string,
  options?: I18n.TranslateOptions | undefined,
) => string = (tag: string, options: I18n.TranslateOptions | undefined = {}) => I18n.t(tag, options);

export const currencies: (
  value: number,
  options?: I18n.ToCurrencyOptions | undefined
) => string = (value: number, options: I18n.ToCurrencyOptions | undefined = {}) => I18n.toCurrency(value, options);

export const numbers: (
  value: number,
  options?: I18n.ToNumberOptions | undefined
) => string = (value: number, options: I18n.ToNumberOptions | undefined = {}) => I18n.toNumber(value, options);
