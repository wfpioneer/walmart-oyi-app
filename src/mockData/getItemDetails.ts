import ItemDetails from '../models/ItemDetails';
/* eslint-disable quote-props */

const floorType = 'Sales Floor';
const salesLastUpdateTs = '2020-07-15T08:02:17-05:00';
const salesDay1 = '2020-07-08';
const salesDay2 = '2020-07-09';
const salesDay3 = '2020-07-10';
const salesDay4 = '2020-07-11';
const salesDay5 = '2020-07-12';
const salesDay6 = '2020-07-13';
const salesDay7 = '2020-07-14';

export default {
  '123': {
    itemName: 'Test Item That is Really, Really Long (and has parenthesis)',
    itemNbr: 1234567890,
    upcNbr: '000055559999',
    status: 'Active',
    categoryNbr: 93,
    categoryDesc: 'Meat PI',
    price: 2000.94,
    basePrice: 1500.30,
    exceptionType: 'nsfl',
    imageUrlKey: undefined,
    imageBlobKey: undefined,
    completed: false,
    onHandsQty: 42,
    pendingOnHandsQty: -999,
    consolidatedOnHandQty: 10,
    claimsOnHandQty: 5,
    backroomQty: 6,
    cloudQty: undefined,
    replenishment: {
      onOrder: 48
    },
    location: {
      floor: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: floorType,
          typeNbr: 8
        },
        {
          zoneId: 0,
          aisleId: 0,
          sectionId: 2,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '2',
          locationName: 'A1-2',
          type: 'End Cap',
          typeNbr: 12
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 3,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '3',
          locationName: 'A1-3',
          type: 'Pod',
          typeNbr: 13
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 4,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '4',
          locationName: 'A1-4',
          type: 'Display',
          typeNbr: 11
        }
      ],
      reserve: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: 'Reserve',
          typeNbr: 7
        }
      ],
      count: 5
    },
    sales: {
      lastUpdateTs: salesLastUpdateTs,
      dailyAvgSales: 15,
      daily: [
        {
          day: salesDay1,
          value: 100
        },
        {
          day: salesDay2,
          value: 0
        },
        {
          day: salesDay3,
          value: 10
        },
        {
          day: salesDay4,
          value: 10
        },
        {
          day: salesDay5,
          value: 42
        },
        {
          day: salesDay6,
          value: 5
        },
        {
          day: salesDay7,
          value: 1
        }
      ],
      weeklyAvgSales: 10,
      weekly: [
        {
          week: 51,
          value: 39
        },
        {
          week: 52,
          value: 26
        },
        {
          week: 53,
          value: 45
        },
        {
          week: 54,
          value: 34
        },
        {
          week: 55,
          value: 34
        }
      ]
    }
  } as ItemDetails,
  '456': {
    itemName: 'Small, Store Use Item',
    itemNbr: 987654321,
    upcNbr: '777555333',
    status: 'Active',
    categoryNbr: 99,
    categoryDesc: 'Store Use',
    price: 2000.94,
    basePrice: 1500.30,
    exceptionType: 'nsfl',
    imageUrlKey: undefined,
    imageBlobKey: undefined,
    completed: false,
    onHandsQty: 42,
    pendingOnHandsQty: -999,
    consolidatedOnHandQty: 0,
    claimsOnHandQty: 5,
    backroomQty: 6,
    cloudQty: 42,
    replenishment: {
      onOrder: 48
    },
    location: {
      floor: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: floorType,
          typeNbr: 8
        },
        {
          zoneId: 0,
          aisleId: 0,
          sectionId: 2,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '2',
          locationName: 'A1-2',
          type: 'End Cap',
          typeNbr: 12
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 3,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '3',
          locationName: 'A1-3',
          type: 'Pod',
          typeNbr: 13
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 4,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '4',
          locationName: 'A1-4',
          type: 'Display',
          typeNbr: 11
        }
      ],
      reserve: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: 'Reserve',
          typeNbr: 7
        }
      ],
      count: 5
    },
    sales: {
      lastUpdateTs: salesLastUpdateTs,
      dailyAvgSales: 15,
      daily: [
        {
          day: salesDay1,
          value: 10
        },
        {
          day: salesDay2,
          value: 0
        },
        {
          day: salesDay3,
          value: 10
        },
        {
          day: salesDay4,
          value: 10
        },
        {
          day: salesDay5,
          value: 42
        },
        {
          day: salesDay6,
          value: 5
        },
        {
          day: salesDay7,
          value: 1
        }
      ],
      weeklyAvgSales: 10,
      weekly: [
        {
          week: 51,
          value: 0
        },
        {
          week: 1,
          value: 10
        },
        {
          week: 2,
          value: 27
        },
        {
          week: 3,
          value: 10
        }
      ]
    }
  } as ItemDetails,
  '789': {
    itemName: 'Wine Item',
    itemNbr: 987654321,
    upcNbr: '777555333',
    status: 'Active',
    categoryNbr: 19,
    categoryDesc: 'Wine',
    price: 2000.94,
    basePrice: 1500.30,
    exceptionType: 'nsfl',
    imageUrlKey: undefined,
    imageBlobKey: undefined,
    completed: false,
    onHandsQty: 42,
    consolidatedOnHandQty: 0,
    claimsOnHandQty: 0,
    backroomQty: 6,
    pendingOnHandsQty: 35,
    cloudQty: 42,
    replenishment: {
      onOrder: 48
    },
    location: {
      floor: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: floorType,
          typeNbr: 8
        },
        {
          zoneId: 0,
          aisleId: 0,
          sectionId: 2,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '2',
          locationName: 'A1-2',
          type: 'End Cap',
          typeNbr: 12
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 3,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '3',
          locationName: 'A1-3',
          type: 'Pod',
          typeNbr: 13
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 4,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '4',
          locationName: 'A1-4',
          type: 'Display',
          typeNbr: 11
        }
      ],
      reserve: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: 'Reserve',
          typeNbr: 7
        }
      ],
      count: 5
    },
    sales: {
      lastUpdateTs: salesLastUpdateTs,
      dailyAvgSales: 15,
      daily: [
        {
          day: salesDay1,
          value: 10
        },
        {
          day: salesDay2,
          value: 0
        },
        {
          day: salesDay3,
          value: 10
        },
        {
          day: salesDay4,
          value: 10
        },
        {
          day: salesDay5,
          value: 42
        },
        {
          day: salesDay6,
          value: 5
        },
        {
          day: salesDay7,
          value: 1
        }
      ],
      weeklyAvgSales: 10,
      weekly: [
        {
          week: 51,
          value: 0
        },
        {
          week: 1,
          value: 10
        },
        {
          week: 2,
          value: 27
        },
        {
          week: 3,
          value: 10
        }
      ]
    }
  } as ItemDetails
};
