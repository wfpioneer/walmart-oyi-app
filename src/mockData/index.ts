
import itemDetails from './getItemDetails';


export const getMockItemDetails = (upc: string) => {
  // @ts-ignore
  return itemDetails[upc] || {};
};
