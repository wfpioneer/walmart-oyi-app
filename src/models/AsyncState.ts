export interface AsyncState{
  isWaiting: boolean;
  value: any;
  error: any;
  result: any;
}
