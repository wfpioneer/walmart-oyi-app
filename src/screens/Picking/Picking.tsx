import React from 'react';
import { FlatList, Text, View } from 'react-native';
import PickItemInfo from '../../components/PickItemInfoCard/PickItemInfoCard';
import { PickListItem, PickStatus } from '../../models/Picking.d';

// TODO Replace with PickTab Screen Component https://jira.walmart.com/browse/INTLSAOPS-5445
const PickingScreen = () => {
  const mockPickListItem: PickListItem[] = [
    {
      assignedAssociate: 'me',
      category: 71,
      createTS: '10:32 AM 02/04/2022',
      createdBy: 'someone else',
      id: 418,
      itemDesc: 'Teapot',
      itemNbr: 734,
      moveToFront: true,
      palletId: 4321,
      palletLocation: 'C1-2',
      quickPick: false,
      salesFloorLocation: 'C1-3',
      status: PickStatus.ACCEPTED_PICK,
      upcNbr: '000041800003'
    }
  ];
  return (
    <View>
      <Text> Picking Screen </Text>
      <FlatList
        data={mockPickListItem}
        keyExtractor={item => item.id.toString()}
        renderItem={({ item }) => (
          <PickItemInfo
            pickListItem={item}
            canDelete={true}
            onDeletePressed={() => undefined}
          />
        )}
      />
    </View>
  );
};

const Picking = () => <PickingScreen />;

export default Picking;
