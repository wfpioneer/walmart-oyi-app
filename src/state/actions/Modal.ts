export const SHOW_ACTIVITY_MODAL = 'MODAL/SHOW_ACTIVITY';
export const HIDE_ACTIVITY_MODAL = 'MODAL/HIDE_ACTIVITY';

export const SHOW_INFO_MODAL = 'MODAL/SHOW_INFO_MODAL';
export const HIDE_INFO_MODAL = 'MODAL/HIDE_INFO_MODAL';

export const showActivityModal = () => ({
  type: SHOW_ACTIVITY_MODAL
});

export const hideActivityModal = () => ({
  type: HIDE_ACTIVITY_MODAL
});

export const showInfoModal = (title: string, text: string) => ({
  type: SHOW_INFO_MODAL,
  payload: { title, text }
});

export const hideInfoModal = () => ({
  type: HIDE_INFO_MODAL
});
