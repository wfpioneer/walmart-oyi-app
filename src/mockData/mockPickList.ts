import { PickListItem, PickStatus } from '../models/Picking.d';

export const mockPickLists: PickListItem[] = [
  {
    assignedAssociate: 'Associate 1',
    category: 71,
    createTS: '10:32 AM 02/04/2022',
    createdBy: 'Associate 1',
    id: 418,
    itemDesc: 'Teapot',
    itemNbr: 734,
    moveToFront: true,
    palletId: 4321,
    palletLocationId: 1672,
    palletLocationName: 'C1-2',
    quickPick: false,
    salesFloorLocationId: 1673,
    salesFloorLocationName: 'C1-3',
    status: PickStatus.ACCEPTED_PICK,
    upcNbr: '000041800003'
  },
  {
    assignedAssociate: 'Associate 1',
    category: 71,
    createTS: '4:25 PM 02/09/2022',
    createdBy: 'Manager 1',
    id: 3,
    itemDesc: 'Swiffer',
    itemNbr: 845,
    moveToFront: true,
    palletId: 5432,
    palletLocationId: 2345,
    palletLocationName: 'BREW1-2',
    quickPick: false,
    salesFloorLocationId: 2345,
    salesFloorLocationName: 'BREW1-2',
    status: PickStatus.ACCEPTED_PICK,
    upcNbr: '000052900004'
  }
];
