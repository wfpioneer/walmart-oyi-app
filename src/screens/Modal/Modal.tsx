import React, { ReactElement, ReactNode } from 'react';
import {
  ActivityIndicator, Modal as OrgModal, Platform, Text, TouchableOpacity, View
} from 'react-native';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import Modal from 'react-native-modal';
import { connect } from 'react-redux';
import COLOR from '../../themes/Color';
import styles from './Modal.style';
import { strings } from '../../locales';
import { hideInfoModal } from '../../state/actions/Modal';

const mapStateToProps = (state: any) => ({
  showModal: state.modal.showModal,
  showActivity: state.modal.showActivity,
  modalContent: state.modal.content
});

const mapDispatchToProps = {
  hideInfoModal
};

interface ActivityModalComponentProps {
  showModal: boolean;
  showActivity: boolean;
  modalContent: any;
  hideInfoModal: Function;
}

class ActivityModalComponent extends React.PureComponent<ActivityModalComponentProps> {
  renderActivityIndicator() {
    if (!this.props.showActivity) {
      return null;
    }
    return (
      <View style={styles.activityView}>
        <ActivityIndicator size="large" color={Platform.OS === 'android' ? COLOR.MAIN_THEME_COLOR : undefined} />
      </View>
    );
  }

  renderContentView() {
    if (!this.props.modalContent) {
      return null;
    }

    return (
      <View style={styles.infoView}>
        <MaterialIcons name="info" size={30} color={COLOR.MAIN_THEME_COLOR} style={styles.normalText} />
        <Text style={styles.titleText}>{this.props.modalContent.title}</Text>
        <Text style={styles.normalText}>{this.props.modalContent.text}</Text>
        <TouchableOpacity style={styles.okButton} onPress={() => this.props.hideInfoModal()}>
          <Text style={styles.okText}>{strings('GENERICS.OK')}</Text>
        </TouchableOpacity>
      </View>
    );
  }

  render() {
    return (
      <Modal isVisible={this.props.showModal} useNativeDriver animationInTiming={0}>
        { this.renderActivityIndicator() }
        { this.renderContentView() }
      </Modal>
    );
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ActivityModalComponent);

interface CustomModalProps {
  children: ReactNode | ReactElement;
  modalType: 'Error' | 'Form' | 'Info'| 'Spinner';
  isVisible: boolean
  onClose: () => void;
}
const stylePicker = {
  Error: styles.errorContainer,
  Form: styles.contentContainer,
  Info: styles.infoView,
  Spinner: styles.activityView
};

export const CustomModal = (props: CustomModalProps): JSX.Element => {
  const {
    children, isVisible, modalType, onClose
  } = props;

  return (
    <OrgModal
      onRequestClose={() => onClose()}
      visible={isVisible}
      transparent={true}
    >
      {modalType === 'Spinner'
        ? (
          <View style={styles.activityView}>
            <ActivityIndicator size="large" color={Platform.OS === 'android' ? COLOR.MAIN_THEME_COLOR : undefined} />
          </View>
        )
        : (
          <View style={styles.modalContainer}>
            <View style={stylePicker[modalType]}>

              {children}
            </View>
          </View>
        )}
    </OrgModal>
  );
};
