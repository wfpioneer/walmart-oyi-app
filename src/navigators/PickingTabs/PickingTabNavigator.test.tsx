import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { NavigationProp, RouteProp } from '@react-navigation/native';
import Toast from 'react-native-toast-message';
import {
  BottomSheetModal
} from '@gorhom/bottom-sheet';
import {
  BottomSheetCard, PickingTabNavigator, getItemDetailsApiHook, getPicklistApiHook, updatePicklistStatusApiHook
} from './PickingTabNavigator';
import { strings } from '../../locales';
import { hideActivityModal, showActivityModal } from '../../state/actions/Modal';
import { AsyncState } from '../../models/AsyncState';
import { mockPickLists } from '../../mockData/mockPickList';
import getItemDetails from '../../mockData/getItemDetails';
import { Tabs } from '../../models/Picking.d';

jest.mock('../../state/actions/Modal', () => ({
  showActivityModal: jest.fn(),
  hideActivityModal: jest.fn()
}));

const defaultAsyncState: AsyncState = {
  isWaiting: false,
  value: null,
  error: null,
  result: null
};
const navigationProp: NavigationProp<any> = {
  addListener: jest.fn(),
  canGoBack: jest.fn(),
  dispatch: jest.fn(),
  goBack: jest.fn(),
  isFocused: jest.fn(() => true),
  removeListener: jest.fn(),
  reset: jest.fn(),
  setOptions: jest.fn(),
  setParams: jest.fn(),
  navigate: jest.fn(),
  getId: jest.fn(),
  getParent: jest.fn(),
  getState: jest.fn()
};
let routeProp: RouteProp<any, string>;

describe('Picking Tab Navigator', () => {
  it('Renders the Pick TabNavigator', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <PickingTabNavigator
        picklist={[]}
        navigation={navigationProp}
        route={routeProp}
        useEffectHook={jest.fn}
        getItemDetailsApi={defaultAsyncState}
        getPicklistsApi={defaultAsyncState}
        updatePicklistStatusApi={defaultAsyncState}
        dispatch={jest.fn()}
        selectedTab={Tabs.PICK}
        useCallbackHook={jest.fn}
        useFocusEffectHook={jest.fn}
        multiBinEnabled={false}
        multiPickEnabled={false}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Renders the Pick TabNavigator with "swipeEnabled: false" if multi Bin/Pick is enabled', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <PickingTabNavigator
        picklist={[]}
        navigation={navigationProp}
        route={routeProp}
        useEffectHook={jest.fn}
        getItemDetailsApi={defaultAsyncState}
        getPicklistsApi={defaultAsyncState}
        updatePicklistStatusApi={defaultAsyncState}
        dispatch={jest.fn()}
        selectedTab={Tabs.PICK}
        useCallbackHook={jest.fn}
        useFocusEffectHook={jest.fn}
        multiBinEnabled={true}
        multiPickEnabled={true}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Renders the Pick TabNavigator with Active Picks', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <PickingTabNavigator
        picklist={mockPickLists}
        navigation={navigationProp}
        route={routeProp}
        useEffectHook={jest.fn}
        getItemDetailsApi={defaultAsyncState}
        getPicklistsApi={defaultAsyncState}
        updatePicklistStatusApi={defaultAsyncState}
        dispatch={jest.fn()}
        selectedTab={Tabs.PICK}
        useCallbackHook={jest.fn}
        useFocusEffectHook={jest.fn}
        multiBinEnabled={false}
        multiPickEnabled={false}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

describe('BottomSheet card', () => {
  it('Renders BottomSheetCard', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <BottomSheetCard
        text="Test"
        onPress={jest.fn()}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

describe('Manage PickingNavigator externalized function tests', () => {
  const mockDispatch = jest.fn();
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Tests getItemDetailsApiHook on 200 success for a new item', () => {
    const successApi: AsyncState = {
      ...defaultAsyncState,
      result: {
        data: getItemDetails[456],
        status: 200
      }
    };
    getItemDetailsApiHook(successApi, mockDispatch, navigationProp);
    expect(navigationProp.navigate).toHaveBeenCalled();
    expect(hideActivityModal).toBeCalledTimes(1);
  });

  it('Tests getItemDetailsApiHook on 204 success for a new item', () => {
    const successApi204: AsyncState = {
      ...defaultAsyncState,
      result: {
        data: '',
        status: 204
      }
    };
    const toastItemNotFound = {
      type: 'error',
      text1: strings('ITEM.ITEM_NOT_FOUND'),
      visibilityTime: 4000,
      position: 'bottom'
    };
    getItemDetailsApiHook(successApi204, mockDispatch, navigationProp);
    expect(mockDispatch).toBeCalledTimes(2);
    expect(Toast.show).toHaveBeenCalledWith(toastItemNotFound);
    expect(hideActivityModal).toBeCalledTimes(1);
  });

  it('Tests getItemDetailsApi on failure', () => {
    const failureApi: AsyncState = {
      ...defaultAsyncState,
      error: 'Internal Server Error'
    };
    const toastGetItemError = {
      type: 'error',
      text1: strings('ITEM.API_ERROR'),
      text2: strings('GENERICS.TRY_AGAIN'),
      visibilityTime: 4000,
      position: 'bottom'
    };
    getItemDetailsApiHook(failureApi, mockDispatch, navigationProp);
    expect(mockDispatch).toBeCalledTimes(2);
    expect(hideActivityModal).toBeCalledTimes(1);
    expect(Toast.show).toHaveBeenCalledWith(toastGetItemError);
  });

  it('Tests getItemDetailsApi isWaiting', () => {
    const isLoadingApi: AsyncState = {
      ...defaultAsyncState,
      isWaiting: true
    };
    getItemDetailsApiHook(isLoadingApi, mockDispatch, navigationProp);
    expect(mockDispatch).toBeCalledTimes(1);
    expect(showActivityModal).toBeCalledTimes(1);
  });

  it('Tests getPicklistApiHook on 200 success', () => {
    const successApi: AsyncState = {
      ...defaultAsyncState,
      result: {
        data: mockPickLists,
        status: 200
      }
    };
    getPicklistApiHook(successApi, mockDispatch, true);
    expect(mockDispatch).toBeCalledTimes(3);
  });

  it('Tests getPicklistApiHook on 204 not found', () => {
    const successApi204: AsyncState = {
      ...defaultAsyncState,
      result: {
        data: '',
        status: 204
      }
    };
    const picklistNotFound = {
      type: 'info',
      text1: strings('PICKING.PICKLIST_NOT_FOUND'),
      visibilityTime: 4000,
      position: 'bottom'
    };

    getPicklistApiHook(successApi204, mockDispatch, true);
    expect(mockDispatch).toBeCalledTimes(3);
    expect(Toast.show).toHaveBeenCalledWith(picklistNotFound);
  });

  it('Tests getPicklistApiHook on error response', () => {
    const failureApi: AsyncState = {
      ...defaultAsyncState,
      error: 'Server Error'
    };

    const picklistError = {
      type: 'error',
      text1: strings('PICKING.PICKLIST_ERROR'),
      text2: strings('GENERICS.TRY_AGAIN'),
      visibilityTime: 4000,
      position: 'bottom'
    };

    getPicklistApiHook(failureApi, mockDispatch, true);
    expect(mockDispatch).toBeCalledTimes(2);
    expect(Toast.show).toHaveBeenCalledWith(picklistError);
  });

  it('Tests updatePicklistStatusApiHook', () => {
    const updateSuccessApi: AsyncState = {
      ...defaultAsyncState,
      result: {
        data: '',
        status: 200
      }
    };
    const updateFailureApi: AsyncState = {
      ...defaultAsyncState,
      error: 'Server Error'
    };
    const updateIsLoadingApi: AsyncState = {
      ...defaultAsyncState,
      isWaiting: true
    };
    // on update success
    updatePicklistStatusApiHook(updateSuccessApi, mockDispatch, true);
    expect(mockDispatch).toBeCalledTimes(3);
    expect(Toast.show).toHaveBeenCalledWith({
      type: 'success',
      text1: strings('PICKING.UPDATE_PICKLIST_STATUS_SUCCESS'),
      visibilityTime: 4000,
      position: 'bottom'
    });

    // on update failure
    mockDispatch.mockReset();
    // @ts-expect-error Reset Toast Object
    Toast.show.mockReset();
    updatePicklistStatusApiHook(updateFailureApi, mockDispatch, true);
    expect(mockDispatch).toBeCalledTimes(2);
    expect(Toast.show).toHaveBeenCalledWith({
      type: 'error',
      text1: strings('PICKING.UPDATE_PICKLIST_STATUS_ERROR'),
      text2: strings('GENERICS.TRY_AGAIN'),
      visibilityTime: 4000,
      position: 'bottom'
    });
    // on api request
    mockDispatch.mockReset();
    updatePicklistStatusApiHook(updateIsLoadingApi, mockDispatch, true);
    expect(mockDispatch).toBeCalledTimes(1);
  });
});
