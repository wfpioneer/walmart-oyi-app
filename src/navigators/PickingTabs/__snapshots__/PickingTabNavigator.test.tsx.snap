// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomSheet card Renders BottomSheetCard 1`] = `
<Memo(BottomSheetView)
  style={
    Object {
      "flexDirection": "row",
      "height": 45,
    }
  }
>
  <TouchableOpacity
    onPress={[MockFunction]}
    style={
      Object {
        "borderColor": "#eeeeee",
        "borderWidth": 1,
        "flexDirection": "row",
        "width": 750,
      }
    }
  >
    <Memo(BottomSheetView)
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
            "paddingLeft": 20,
          }
        }
      >
        Test
      </Text>
    </Memo(BottomSheetView)>
  </TouchableOpacity>
</Memo(BottomSheetView)>
`;

exports[`Picking Tab Navigator Renders the Pick TabNavigator 1`] = `
<MaterialTopTabNavigator
  initialRouteName="Pick"
  screenListeners={
    Object {
      "beforeRemove": [Function],
    }
  }
  screenOptions={
    Object {
      "tabBarActiveTintColor": "#0071e9",
      "tabBarInactiveTintColor": "#616161",
      "tabBarItemStyle": Object {
        "borderColor": "#9e9e9e",
        "borderEndWidth": 0.5,
      },
    }
  }
>
  <Screen
    listeners={
      Object {
        "focus": [Function],
      }
    }
    name="QuickPick"
    options={
      Object {
        "tabBarBadge": undefined,
        "title": "[missing \\"en.PICKING.QUICKPICK\\" translation]",
      }
    }
  >
    [Function]
  </Screen>
  <Screen
    listeners={
      Object {
        "focus": [Function],
      }
    }
    name="Pick"
    options={
      Object {
        "tabBarBadge": undefined,
        "title": "[missing \\"en.PICKING.PICK\\" translation]",
      }
    }
  >
    [Function]
  </Screen>
  <Screen
    listeners={
      Object {
        "focus": [Function],
      }
    }
    name="SalesFloor"
    options={
      Object {
        "tabBarBadge": undefined,
        "title": "[missing \\"en.ITEM.SALES_FLOOR_QTY\\" translation]",
      }
    }
  >
    [Function]
  </Screen>
</MaterialTopTabNavigator>
`;

exports[`Picking Tab Navigator Renders the Pick TabNavigator with Active Picks 1`] = `
<MaterialTopTabNavigator
  initialRouteName="Pick"
  screenListeners={
    Object {
      "beforeRemove": [Function],
    }
  }
  screenOptions={
    Object {
      "tabBarActiveTintColor": "#0071e9",
      "tabBarInactiveTintColor": "#616161",
      "tabBarItemStyle": Object {
        "borderColor": "#9e9e9e",
        "borderEndWidth": 0.5,
      },
    }
  }
>
  <Screen
    listeners={
      Object {
        "focus": [Function],
      }
    }
    name="QuickPick"
    options={
      Object {
        "tabBarBadge": undefined,
        "title": "[missing \\"en.PICKING.QUICKPICK\\" translation]",
      }
    }
  >
    [Function]
  </Screen>
  <Screen
    listeners={
      Object {
        "focus": [Function],
      }
    }
    name="Pick"
    options={
      Object {
        "tabBarBadge": [Function],
        "title": "[missing \\"en.PICKING.PICK\\" translation]",
      }
    }
  >
    [Function]
  </Screen>
  <Screen
    listeners={
      Object {
        "focus": [Function],
      }
    }
    name="SalesFloor"
    options={
      Object {
        "tabBarBadge": undefined,
        "title": "[missing \\"en.ITEM.SALES_FLOOR_QTY\\" translation]",
      }
    }
  >
    [Function]
  </Screen>
</MaterialTopTabNavigator>
`;
